plugins/modules/web_infrastructure/jira.py

```diff
@@ -388,6 +406,10 @@ def comment(restbase, user, passwd, params):
         'body': params['comment']
     }
 
+    # if comment_visibility is specified restrict visibility
+    if 'comment_visibility' in params.keys():
```

Comment:
That condition is *always* true. You should rather do:
```suggestion
    if params['comment_visibility'] is not None:
```